apps/server/src/form-instances/form-instances.service.ts

```diff
@@ -99,4 +99,31 @@ export class FormInstancesService {
       },
     });
   }
+
+  async signFormInstance(formInstanceId: string, signatureId: string) {
+    const formInstance = await this.prisma.formInstance.findUnique({
+        where: { id: formInstanceId },
+        include: { signatures: { include: { signerPosition: true, userSignedBy: true } } },
+    });
+
+    if (!formInstance) {
+        throw new Error('Form instance not found');
+    }
+
+    const signature = formInstance.signatures.find(sig => sig.id === signatureId);
```

Comment:
we should verify that the signature is the next one to be signed (all previous signatures should be signed), or else throw an error